packages/dialtone-vue2/components/rich_text_editor/rich_text_editor.vue

```diff
@@ -578,6 +568,26 @@ export default {
       this.addEditorListeners();
     },
 
+    processValue (newValue) {
+      let currentValue = this.getOutput();
+      if (this.outputFormat === 'json') {
+        newValue = JSON.stringify(newValue);
+        currentValue = JSON.stringify(currentValue);
+      }
+      if (newValue === currentValue) {
+        // The new value came from this component and was passed back down
+        // through the parent, so don't do anything here.
+        return;
+      }
+
+      if (this.outputFormat === 'text') {
+        // If the text contains emoji characters convert them to emoji component tags
+        newValue = newValue.replace(/(\p{Emoji}\p{Emoji_Modifier}?)/gu, '<emoji-component code="$1"></emoji-component>');
```

Comment:
What does `/(\p{Emoji}\p{Emoji_Modifier}?)/gu` regex do?

Reply:
Unicode character escape to match all emoji within the string. `Emoji_Modifier` is for the optional skin-tone modifier.

Reply:
Nice find.. seems there are some strange gaps when using this `{emoji}` method that don't work. I will use a different method.